database/chain.go

```diff
@@ -0,0 +1,429 @@
+	chainSwap.Status = status
+
+	_, err := database.Exec("UPDATE chainSwaps SET status = ? WHERE id = ?", status.String(), chainSwap.Id)
+	return err
+}
+
+func (database *Database) SetChainSwapLockupTransactionId(swapData *ChainSwapData, lockupTransactionId string) error {
+	swapData.LockupTransactionId = lockupTransactionId
+	_, err := database.Exec("UPDATE chainSwapsData SET lockupTransactionId = ? WHERE id = ? AND currency = ?", lockupTransactionId, swapData.Id, swapData.Currency)
+	return err
+}
+
+func (database *Database) SetChainSwapTransactionId(swapData *ChainSwapData, transactionId string) error {
+	swapData.Transactionid = transactionId
+	_, err := database.Exec("UPDATE chainSwapsData SET transactionId = ? WHERE id = ? AND currency = ?", transactionId, swapData.Id, swapData.Currency)
+	return err
+}
+
+func (database *Database) AddChainSwapOnchainFee(chainSwap *ChainSwap, onchainFee uint64) error {
+	chainSwap.OnchainFee = addToOptional(chainSwap.OnchainFee, onchainFee)
+	_, err := database.Exec("UPDATE chainSwaps SET onchainFee = ? WHERE id = ?", chainSwap.OnchainFee, chainSwap.Id)
+	return err
+}
+
+func (database *Database) SetChainSwapServiceFee(chainSwap *ChainSwap, serviceFee uint64) error {
+	chainSwap.ServiceFee = &serviceFee
+	_, err := database.Exec("UPDATE chainSwaps SET serviceFee = ? WHERE id = ?", chainSwap.ServiceFee, chainSwap.Id)
+	return err
+}
+
+func (database *Database) UpdateChainSwapState(chainSwap *ChainSwap, state boltzrpc.SwapState, error string) error {
+	chainSwap.State = state
+	chainSwap.Error = error
+
+	_, err := database.Exec("UPDATE chainSwaps SET state = ?, error = ? WHERE id = ?", state, error, chainSwap.Id)
+	return err
+}
+
+func (database *Database) QueryChainSwap(id string) (swap *ChainSwap, err error) {
+	database.lock.RLock()
+	defer database.lock.RUnlock()
+	rows, err := database.Query("SELECT * FROM chainSwaps WHERE id = ?", id)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	if rows.Next() {
+		swap, err = database.parseChainSwap(rows)
+
+		if err != nil {
+			return swap, err
+		}
+	} else {
+		return swap, errors.New("could not find Swap " + id)
+	}
+
+	return swap, err
+}
+
+func (database *Database) QueryChainSwapData(id string, currency boltz.Currency, isClaim bool) (data *ChainSwapData, err error) {
+	database.lock.RLock()
+	defer database.lock.RUnlock()
+	rows, err := database.Query("SELECT * FROM chainSwapsData WHERE id = ? AND currency = ?", id, currency)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	if rows.Next() {
+		data, err = parseChainSwapData(rows, isClaim)
+
+		if err != nil {
+			return data, err
+		}
+	} else {
+		return data, errors.New("could not find data " + id)
+	}
+
+	return data, nil
+}
+
+const refundableSwapsQuery = `
+SELECT swaps.*
+FROM chainSwaps swaps
+         JOIN chainSwapsData data ON swaps.id = data.id AND data.currency = swaps.fromCurrency
+WHERE (state = ? OR state = ? OR state = ?)
```

Comment:
```suggestion
WHERE (state = ? OR state = ? OR state = ?)
```
`IN` would be more elegant https://www.sqlitetutorial.net/sqlite-in/